nats.go

```diff
@@ -4410,6 +4412,65 @@ func (s *Subscription) Drain() error {
 	return conn.unsubscribe(s, 0, true)
 }
 
+type DrainStatus interface {
+	PendingMsgs() int
+	Draining() bool
+	Complete() <-chan struct{}
+}
+
+type drainStatus struct {
+	sub *Subscription
+}
+
+func (s *Subscription) DrainStatus() DrainStatus {
+	s.mu.Lock()
+	defer s.mu.Unlock()
+	return &drainStatus{
+		sub: s,
+	}
+}
+
+func (s *drainStatus) Draining() bool {
```

Comment:
could be part of subscription state instead `sub.IsDraining`  maybe? like the sub companion for `nc.IsDraining`